Duckling/Time/EN/Rules.hs

```diff
@@ -1392,12 +1392,22 @@ ruleIntervalFrom = Rule
   , pattern =
     [ regex "(from|since|start(s|ing)?( in| on)?)"
     , dimension Time
-    , regex "\\-|to|th?ru|through|(un)?til(l)?|(and )?end(s|ing)?( in| on)?"
+    , regex "((((\\-|to|th?ru|through|(un)?til(l)?)((( the)?( end(s|ing)?)? of( the month( of)?)?)?))|(and )?end(s|ing)?( in| on)?))"
     , dimension Time
     ]
   , prod = \tokens -> case tokens of
-      (_:Token Time td1:_:Token Time td2:_) ->
-        Token Time <$> interval TTime.Closed td1 td2
+      (_:Token Time start:Token RegexMatch (GroupMatch (match:_)):Token Time end:_)
+        | (Just start, Just end) <- parsed ->
+          Token Time <$> interval TTime.Open start end
```

Comment:
Converting this interval from `Closed` to `Open` has the side effect of making the result exclusive of the end time instead of inclusive:

Example: "from June through July"

master:
```json
"to": {
    "value": "2022-08-01T00:00:00.000+00:00",
    "grain": "month"
},
"from": {
    "value": "2022-06-01T00:00:00.000+00:00",
    "grain": "month"
},
```

this branch:
```json
"to": {
    "value": "2022-07-01T00:00:00.000+00:00",
    "grain": "month"
},
"from": {
    "value": "2022-06-01T00:00:00.000+00:00",
    "grain": "month"
},
```

More info on `Closed` vs `Open`:
```haskell
-- | Closed if the interval between A and B should include B
-- Open if the interval should end right before B
data TimeIntervalType = Open | Closed
  deriving (Eq, Show)
```